mobile/src/main/java/butter/droid/ui/trailer/TrailerPlayerActivity.java

```diff
@@ -0,0 +1,208 @@
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Butter is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Butter. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package butter.droid.ui.trailer;
+
+import android.app.AlertDialog;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.content.Intent;
+import android.net.Uri;
+import android.os.AsyncTask;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.MenuItem;
+import butter.droid.MobileButterApplication;
+import butter.droid.R;
+import butter.droid.base.manager.network.NetworkManager;
+import butter.droid.base.manager.phone.PhoneManager;
+import butter.droid.base.manager.youtube.YouTubeManager;
+import butter.droid.base.providers.media.models.Media;
+import butter.droid.base.torrent.StreamInfo;
+import butter.droid.base.torrent.TorrentService;
+import butter.droid.fragments.VideoPlayerFragment;
+import butter.droid.ui.ButterBaseActivity;
+import java.lang.ref.WeakReference;
+import java.net.URLDecoder;
+import javax.inject.Inject;
+import timber.log.Timber;
+
+public class TrailerPlayerActivity extends ButterBaseActivity implements TrailerPlayerView, VideoPlayerFragment.Callback {
+
+    public final static String LOCATION = "stream_url";
+    public final static String DATA = "video_data";
+
+    private static final String TAG = TrailerPlayerActivity.class.getSimpleName();
+
+    @Inject
+    TrailerPlayerPresenter presenter;
+    @Inject
+    YouTubeManager youTubeManager;
+    @Inject
+    NetworkManager networkManager;
+    @Inject
+    PhoneManager phoneManager;
+
+    private VideoPlayerFragment videoPlayerFragment;
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        MobileButterApplication.getAppContext()
+                .getComponent()
+                .trailerComponentBuilder()
+                .trailerModule(new TrailerPlayerModule(this))
+                .build()
+                .inject(this);
+
+        super.onCreate(savedInstanceState, R.layout.activity_videoplayer);
+
+        final Media media = getIntent().getParcelableExtra(DATA);
+        media.title += " " + getString(R.string.trailer);
+        final String youtubeUrl = getIntent().getStringExtra(LOCATION);
+
+        this.videoPlayerFragment = (VideoPlayerFragment) getSupportFragmentManager().findFragmentById(R.id.video_fragment);
+
+        presenter.onCreate(media, youtubeUrl);
+    }
+
+    public static Intent getIntent(Context context, Media media, String url) {
```

Comment:
we don't force order in our settings. Only thing that is specified is where auto generated methods should go. If you do it manually you should take care of this yourself.